packages/xrpl/src/Wallet/index.ts

```diff
@@ -466,6 +466,22 @@ class Wallet {
       txCopy.URI = txCopy.URI.toUpperCase()
     }
 
+    if (
+      txCopy.TransactionType === 'NFTokenMint' &&
+      typeof txCopy.URI === 'string' &&
+      txCopy.URI === ''
+    ) {
+      throw new ValidationError('NFTokenMint: URI must not be empty string')
+    }
```

Comment:
This check shouldn't be happening here - I believe `validate` is already run on the transaction before this point.